keps/sig-api-machinery/3285-subresource-finalizers/README.md

```diff
@@ -0,0 +1,537 @@
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes a way to set different RBACs to `finalizers` field and its entire resource.
+
+## Motivation
+
+`Finalizers` is used by controllers as a pre-deletion hook.
+Currently, `finalizers` field isn't subresourced, therefore RBAC for the field can only be set through the one for its entire resource.
+On the other hand, some controllers only need to update `finalizers` field and some of its subresources, and don't need to update its entire resource.
+To implement least privilege for such controllers, there should be a way to only allow updating `finalizers`.
+
+### Goals
+
+- Provide a way to set a different RBAC to `finalizers` from its entire resource
+
+### Non-Goals
+
+- Provide a way to set a different RBAC to other than `finalizers` from its entire resource
+- Provide a way to allow updating its entire resource  but disallow updating `finalizers` field
+
+## Proposal
+
+A new endpoint `finalize` is added as an alternative way to access to `finalizers` field and allow setting a separate RBAC to the endpoint.
```

Comment:
Resources and subresources are usually nouns or noun-like. This should be `finalizers` not `finalize`.

Compare `status`, `tokenrequest`, `scale`: all noun-like.

Reply:
status is "literally a sub-section of the type".  scale is ambiguous.  You got me with "tokenrequest", though.  Whatever we pick, we can fine-tune through impl and alpha - the people I expect to have opinions here are AFK for a bit :)

Reply:
I found that [the name `finalize` is taken](https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/rest/storage_core.go#L344-L347) for another use. So, I need another good name for this endpoint. Any ideas? (Just `finalizer` or `finalizers`?)

Reply:
that is super unfortunate.  I don't know what that is for, either.